src/libraries/Common/src/System/Security/Cryptography/MLDsaImplementation.cs

```diff
@@ -17,10 +17,32 @@ private MLDsaImplementation(MLDsaAlgorithm algorithm)
 
         internal static partial bool SupportsAny();
 
-        internal static partial MLDsa GenerateKeyImpl(MLDsaAlgorithm algorithm);
-        internal static partial MLDsa ImportPublicKey(MLDsaAlgorithm algorithm, ReadOnlySpan<byte> source);
-        internal static partial MLDsa ImportPkcs8PrivateKeyValue(MLDsaAlgorithm algorithm, ReadOnlySpan<byte> source);
-        internal static partial MLDsa ImportSecretKey(MLDsaAlgorithm algorithm, ReadOnlySpan<byte> source);
-        internal static partial MLDsa ImportSeed(MLDsaAlgorithm algorithm, ReadOnlySpan<byte> source);
+        internal static partial MLDsaImplementation GenerateKeyImpl(MLDsaAlgorithm algorithm);
+        internal static partial MLDsaImplementation ImportPublicKey(MLDsaAlgorithm algorithm, ReadOnlySpan<byte> source);
+        internal static partial MLDsaImplementation ImportPkcs8PrivateKeyValue(MLDsaAlgorithm algorithm, ReadOnlySpan<byte> source);
+        internal static partial MLDsaImplementation ImportSecretKey(MLDsaAlgorithm algorithm, ReadOnlySpan<byte> source);
+        internal static partial MLDsaImplementation ImportSeed(MLDsaAlgorithm algorithm, ReadOnlySpan<byte> source);
+
+        internal static MLDsaImplementation DuplicatePrivateKey(MLDsa key)
```

Comment:
Should this be a partial and each `MLDsaImplementation` knows how to duplicate itself? If this is partial, then the `MLDsaImplementation.OpenSsl` implemented partial can duplicate the EVP_PKEY (and Windows can/will do whatever Windows needs to do).

Reply:
Now that I have gotten to `MLDsaImplementation.OpenSsl.cs`, I guess I don't understand why we need both `DuplicatePrivateKey` and `FromHandle`.

Reply:
This is for the "MLDsaOther" case.  CopyWithPrivateKey agrees that the public keys match, but doesn't understand the object, so needs to convert it to something it does.

The current algorithms do this, too, but inline (and copy/pasted to each PAL), e.g. https://github.com/dotnet/runtime/blob/b1d93d078659f71068715b5fc5a1843d64395490/src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/OpenSslX509CertificateReader.cs#L689-L700

This time I had the insight that everyone inbox should understand what they need from the imol type... though I guess that is debatable for Windows... the impl will have a BCrypt handle but GetPrivateKey should be NCrypt. Hmm.

Reply:
Added the assert, and some comments